### PR TITLE
Hot deploy for Vertx

### PR DESCRIPTION
**Pull request by g4s8**
#201 - added hot deploy Vertx Maven plugin, implemented `Verticle` for main class.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#208](https://github.com/artipie/artipie/issues/208) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/ac34c0cb-eaf5-47e6-b956-6f18039fa194); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/VertxMain.java [hunk 1]
```diff
@@ -106,16 +88,57 @@ public static void main(final String... args) throws IOException, ParseException
...
-            vertx,
-            port
-        ).run();
+        final Context ctx = vertx.getOrCreateContext();
+        final JsonObject config = ctx.config();
+        config.put("storage", storage);
+        config.put("port", port);
+        final VertxMain main = new VertxMain();
+        main.init(vertx, ctx);
+        main.start(
```
Comment (Sammers21) [inline src/main/java/com/artipie/VertxMain.java:97]: The `start` method is not supposed to be called manually. Read the documentation: https://vertx.io/docs/vertx-core/java/#_verticles

Comment (Sammers21) [reply src/main/java/com/artipie/VertxMain.java:97]: You create a verticle, and then deploy it:
```java
Verticle myVerticle = new MyVerticle();
vertx.deployVerticle(myVerticle);
```

File: src/main/resources/debug-config.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (Sammers21) [inline src/main/resources/debug-config.json:1]: Plugin documentation mentions that "verticleReference" field is required

**Comment by 0crat** [comment]
@vatavuk[/z](https://www.0crat.com/u/vatavuk) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by Vatavuk** [comment]
@g4s8 ping

**Comment by Vatavuk** [comment]
@g4s8 ping

**Comment by Vatavuk** [comment]
@g4s8 ping

**Comment by Vatavuk** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/artipie/issues/208#issuecomment-654725697))

@Vatavuk The user @vatavuk[/z](https://www.0crat.com/u/vatavuk) resigned from [#208](https://github.com/artipie/artipie/issues/208), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
This pull request [#208](https://github.com/artipie/artipie/issues/208) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/b892067a-9844-4ef7-a3db-395d845f4abf); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by carlosmiranda** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/artipie/issues/208#issuecomment-655404820))

@carlosmiranda The user @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) resigned from [#208](https://github.com/artipie/artipie/issues/208), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/artipie/issues/208#issuecomment-655404820))

@carlosmiranda Job refused in 20 hours - no penalty, see [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
This pull request [#208](https://github.com/artipie/artipie/issues/208) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/63a3c216-ae6f-42a6-a0b3-ca3cd7914279); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by paulodamaso** [comment]
@g4s8 PLease solve conflicts and assess the comments made by @Sammers21 and @Vatavuk

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by paulodamaso** [comment]
@g4s8 ping

**Comment by paulodamaso** [comment]
@g4s8 please solve conflicts

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR

**Comment by 0crat** [comment]
Pull request [#208](https://github.com/artipie/artipie/issues/208) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by 0crat** [comment]
The project doesn't have enough funds, can't make a payment

**Comment by 0crat** [comment]
Code review was too long (35 days), architects (@g4s8, @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova)) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Pull request closed without merging**